test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/random/GeneratorUtilTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.services.test.utils.core.random;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.checkValidityOfText;
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.chooseRandomPosition;
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.concat;
+
+import java.util.Random;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class GeneratorUtilTest {
+
+    @Test
+    public void shoulGenerateValidStringFromAListOfCharacters() {
+        char a[] = "abcef".toCharArray();
+        String gen = GeneratorUtil.generateStringFromCharacters(new Random(), a, 2, 3);
+        for (String c : gen.split("")) {
+            assertTrue("abcef".contains(c));
+        }
+    }
+
+    @Test
+    public void shouldChooseRandomPosition() {
+        for (int i = 1; i < 10000; i++) {
+            int chosen = chooseRandomPosition(i);
+            assertTrue(chosen < i);
+            assertTrue(chosen >= 0);
+        }
+    }
+
+    @Test
+    public void shouldConcatenateTwoCharacterArrays() {
+        char a[] = "abcef".toCharArray();
+        char b[] = "abcef".toCharArray();
+        char c[] = concat(a, b);
+        Assert.assertArrayEquals("abcefabcef".toCharArray(), c);
+    }
+
+    @Test
+    public void shouldTestThatStringIsComprisedOfValidCharacters() {
+        char a[] = "abcef".toCharArray();
+        char b[] = "abcef".toCharArray();
+        assertTrue(checkValidityOfText(new String(a), b));
+        assertFalse(checkValidityOfText(new String(a), "ghij".toCharArray()));
+    }
+
```

Comment:
This test is likely to fail randomly. Just because it is a 1/100 probablilyt of being true, doesn't mean that once in every hundred times is true. It could run 1000 times and still not return true (1/10 probability)